molgenis-data/src/test/java/org/molgenis/data/meta/model/AttributeTest.java

```diff
@@ -24,14 +24,65 @@
 import static org.testng.Assert.assertNull;
 import static org.testng.Assert.assertTrue;
 
-import org.testng.annotations.BeforeMethod;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.molgenis.data.Sort;
+import org.molgenis.data.config.EntityBaseTestConfig;
+import org.molgenis.data.config.MetadataTestConfig;
+import org.molgenis.data.meta.AbstractSystemEntityTest;
+import org.molgenis.util.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
 import org.testng.annotations.Test;
 
-public class AttributeTest {
+@ContextConfiguration(
+    classes = {
+      EntityBaseTestConfig.class,
+      AttributeMetadata.class,
+      AttributeFactory.class,
+      MetadataTestConfig.class
+    })
+public class AttributeTest extends AbstractSystemEntityTest {
+
+  @Autowired AttributeMetadata metadata;
+  @Autowired AttributeFactory factory;
+
+  @Override
+  protected List<String> getExcludedAttrs() {
+    List<String> attrs = new ArrayList<>();
+    attrs.add(AttributeMetadata.MAPPED_BY);
+    attrs.add(AttributeMetadata.ENUM_OPTIONS);
+    return attrs;
+  }
+
+  @Override
+  protected Map<String, Pair<Class, Object>> getOverriddenReturnTypes() {
+    Pair<Class, Object> statusPair = new Pair<>();
+    statusPair.setA(Sort.class);
+    statusPair.setB(Sort.parse("attr"));
+
+    Pair<Class, Object> enumOptions = new Pair<>();
+    enumOptions.setA(List.class);
+    enumOptions.setB(Collections.singletonList("test"));
+
+    Map<String, Pair<Class, Object>> map = new HashMap<>();
+    map.put(AttributeMetadata.ORDER_BY, statusPair);
```

Comment:
What about the `enumOptions`?

Reply:
excluded, If I remember correct the getter and setter do not match

Reply:
I mean, they're being defined a few lines above this one but not used:

```java 
   Pair<Class, Object> enumOptions = new Pair<>();
   enumOptions.setA(List.class);
   enumOptions.setB(Collections.singletonList("test"));
```

Reply:
ah, tried to test it via this, failed, moved it to "excluded" and didn't clean up properly